File: java/AbstractFactory/src/absfactory/PhoneNumberFactory.java
package absfactory;

public class PhoneNumberFactory implements AbstractFactory{

	@Override
	public PostCode getPostCode(Country country, String postalCode) {
		return null;
	}

	@Override
	public PhoneNumber getPhoneNumber(Country country, String phoneNumber) {
		if (country == Country.US)
			return new USPhoneNumber(phoneNumber);
		if (country == Country.UK)
			return new UKPhoneNumber(phoneNumber);
		
		return null;
	}
}

File: ruby/AbstractFactory/phone_number_factory.rb
require_relative 'us_phone_number'
require_relative 'uk_phone_number'

class PhoneNumberFactory
  def get_instance(country, phone_number)
    if country == "US"
      return USPhoneNumber.new(phone_number)
    elsif country == "UK"
      return UKPhoneNumber.new(phone_number)
    end

    raise 'Country is not supported'
  end
end